cmd/query/app/http_handler.go

```diff
@@ -454,7 +454,15 @@ func (aH *APIHandler) writeJSON(w http.ResponseWriter, r *http.Request, response
 			return json.MarshalIndent(v, "", "    ")
 		}
 	}
-	resp, _ := marshall(response)
+	resp, err := marshall(response)
+	if err != nil {
+		aH.handleError(w, fmt.Errorf("failed marshalling HTTP response to JSON: %w", err), http.StatusInternalServerError)
+		return
+	}
 	w.Header().Set("Content-Type", "application/json")
-	w.Write(resp)
+	if b, err := w.Write(resp); err != nil {
+		aH.handleError(w, fmt.Errorf("failed writing HTTP response: %w", err), http.StatusInternalServerError)
+	} else {
+		aH.logger.Debug("Successfully wrote HTTP response", zap.Int("bytes", b))
```

Comment:
I'm not a fan of this debug logging, is it necessary? It should be observable from the client side already.